src/hotspot/share/gc/z/zPageAllocator.cpp

```diff
@@ -1373,9 +1373,25 @@ size_t ZPageAllocator::unused() const {
   return unused > 0 ? (size_t)unused : 0;
 }
 
-ZPageAllocatorStats ZPageAllocator::stats(ZGeneration* generation) const {
-  ZLocker<ZLock> locker(&_lock);
+void ZPageAllocator::update_collection_stats(ZGenerationId id) {
+  assert(SafepointSynchronize::is_at_safepoint(), "Should be at safepoint");
 
+#ifdef ASSERT
+  size_t total_used = 0;
+
+  ZPartitionIterator iter(&_partitions);
+  for (ZPartition* partition; iter.next(&partition);) {
+    total_used += partition->_used;
+  }
+
+  assert(total_used == _used, "Must be consistent at safepoint %zu == %zu", total_used, _used);
```

Comment:
Preexisting, the assert message is misleading. Currently it is not the safepoint which guarantees consistency, but the page allocator lock. 

I wrote this assert at first under the assumption that we did not change _used concurrently with safepoints, but we did, so added the lock but forgot to update the assert text.


Maybe just removing the mention of safepoint is enough.
```suggestion
  assert(total_used == _used, "Must be consistent %zu == %zu", total_used, _used);
```